appinventor/blocklyeditor/src/blocks/components.js

```diff
@@ -87,11 +102,13 @@ Blockly.Blocks.component_event = {
     // [lyn, 12/23/2013] Move this out of domToMutation into top-level component_event
     // this.onchange = Blockly.WarningHandler.checkErrors;
 
-    if (this.getEventTypeObject().deprecated === "true" && this.workspace === Blockly.mainWorkspace) {
+    if (this.getEventTypeObject() && this.getEventTypeObject().deprecated === "true" && this.workspace === Blockly.mainWorkspace) {
       this.badBlock();
       this.setDisabled(true);
     }
 
+    this.verify(); // verify the block and mark it accordingly
```

Comment:
Every component's domToMutation() function, tries to heal a damaged or undefined block if it can.
After trying to heal it, (not always possible) it tries to verify the block by calling this.verify().